extension/src/Documentation.ts

```diff
@@ -1234,6 +1186,264 @@ export async function generateExternalDocumentation(
         saveContentToFile(tocFilepath, tocContent);
       }
     }
+
+    function getTableFieldsTable(object: ALObject): string {
+      let objectIndexContent = "";
+      const fields = (object.controls.filter(
+        (o) => o.type === ALControlType.tableField
+      ) as ALTableField[]).filter(
+        (o) => !o.isObsoletePending() && !o.isObsolete()
+      );
+      if (fields.length > 0) {
+        const printSummary =
+          fields.find((x) => x.xmlComment?.summary !== undefined) !== undefined;
+
+        objectIndexContent += `## Fields\n\n`;
+        objectIndexContent += `| Number | Name | Type |${
+          printSummary ? " Description |" : ""
+        }\n`;
+        objectIndexContent += `| ---- | ------- | ----------- |${
+          printSummary ? " ------------- |" : ""
+        }\n`;
+        fields.forEach((field) => {
+          objectIndexContent += `| ${field.id} | ${field.name} | ${
+            field.dataType
+          } |${
+            printSummary
+              ? ` ${
+                  field.xmlComment?.summary
+                    ? ALXmlComment.formatMarkDown({
+                        text: field.xmlComment.summary,
+                        inTableCell: true,
+                      })
+                    : ""
+                } |`
+              : ""
+          }\n`;
+        });
+        objectIndexContent += "\n";
+      }
+      return objectIndexContent;
+    }
+
+    function getPageFieldsTable(
+      object: ALObject,
+      ignoreTransUnitsSetting: string[],
+      settings: Settings
+    ): string {
+      let objectIndexContent = "";
+      let controls = getAlControlsToPrint(object, ignoreTransUnitsSetting);
+      controls = controls.filter((c) => !c.isObsoletePending(false));
+
+      if (controls.length > 0) {
```

Comment:
is this length check really necessary? Since almost all of the code is within the `forEach`.

Reply:
The same probably goes for some the other functions as well.

Reply:
Yes, nice find!
They where probably necessary once upon the time... 🙂